src/components/SurveyQuestion/RatingQuestion.tsx

```diff
@@ -0,0 +1,113 @@
+import Image from 'next/image';
+import { useState } from 'react';
+
+import { SurveyQuestionProps } from './';
+
+interface RatingItemProps {
+  index: number;
+  ratingType?: string;
+  answerClass: string;
+  onClick: () => void;
+  onMouseEnter: () => void;
+  onMouseLeave: () => void;
+}
+
+const answerClass = (
+  answerIndex: number,
+  selectedIndex: number,
+  hoverIndex: number
+) => {
+  const isActiveAnswer = answerIndex <= Math.max(selectedIndex, hoverIndex);
+
+  if (isActiveAnswer) {
+    return 'opacity-100';
+  }
+
+  return 'opacity-50';
+};
+
+const answerIcon = (ratingType?: string) => {
+  switch (ratingType) {
+    case 'smiley':
+      return '/icon/smiley.svg';
+    case 'star':
+      return '/icon/star.svg';
+    case 'heart':
+      return '/icon/heart.svg';
+    default:
+      return '/icon/star.svg';
+  }
+};
+
+const RaingItem = ({
```

Comment:
```suggestion
const RatingItem = ({
```

By the way, should we move it into its own file? 🤔 So it would makes the split more practical~

Reply:
Updated in [4656b88](https://github.com/hanam1ni/react-surveyor/pull/79/commits/4656b885ba672e3063fac0f9624bf2fee855cf25) alongside conflicts fixing